vtl-spark/src/main/java/fr/insee/vtl/spark/SparkProcessingEngine.java

```diff
@@ -198,16 +200,103 @@ public DatasetExpression executeUnion(List<DatasetExpression> datasets) {
+        AbstractMap.SimpleImmutableEntry<String, String> a2 = new AbstractMap.SimpleImmutableEntry<String, String>("weight", "avgWeight");
+        AbstractMap.SimpleImmutableEntry<String, String> a3 = new AbstractMap.SimpleImmutableEntry<String, String>("null", "countVal");
+        AbstractMap.SimpleImmutableEntry<String, String> a4 = new AbstractMap.SimpleImmutableEntry<String, String>("age", "maxAge");
+        AbstractMap.SimpleImmutableEntry<String, String> a5 = new AbstractMap.SimpleImmutableEntry<String, String>("weight", "maxWeight");
+        AbstractMap.SimpleImmutableEntry<String, String> a6 = new AbstractMap.SimpleImmutableEntry<String, String>("age", "minAge");
+        AbstractMap.SimpleImmutableEntry<String, String> a7 = new AbstractMap.SimpleImmutableEntry<String, String>("weight", "minWeight");
+        AbstractMap.SimpleImmutableEntry<String, String> a8 = new AbstractMap.SimpleImmutableEntry<String, String>("age", "medianAge");
+        AbstractMap.SimpleImmutableEntry<String, String> a9 = new AbstractMap.SimpleImmutableEntry<String, String>("weight", "medianWeight");
+
+        aliases.add(a1);
+        aliases.add(a2);
+        aliases.add(a3);
+        aliases.add(a4);
+        aliases.add(a5);
+        aliases.add(a6);
+        aliases.add(a7);
+        aliases.add(a8);
+        aliases.add(a9);
+
+
         return executeAggr(
                 asSparkDataset(dataset),
-                List.of("foo", "bar"),
-                Map.of("col1", "sum", "col2", "avg"),
-                Map.of("col1", "alias1", "col2", "alias2"));
+                List.of("country"),
+                operations,
+                aliases);
     }
 
-    public DatasetExpression executeAggr(SparkDataset dataset, List<String> groupByColumns,
-                                         Map<String, String> operations, Map<String, String> aliases) {
-        throw new UnsupportedOperationException("TODO");
+    public DatasetExpression executeAggr(SparkDataset dataset, List<String> groupByColNames,
+                                         List<AbstractMap.SimpleImmutableEntry<String, String>> operations,
+                                         List<AbstractMap.SimpleImmutableEntry<String, String>> aliases) {
+        /*
+         * - agg(java.util.Map<String,String> exprs)
+         * - agg(Column expr, scala.collection.Seq<Column> exprs)
+         * */
+        Dataset<Row> df = dataset.getSparkDataset();
+
+        SparkAggrFuncExprBuilder builder = null;
+        try {
+            builder = new SparkAggrFuncExprBuilder(operations, aliases);
+        } catch (Exception e) {
+            // log exception
+            // To do handle it here or propagate the exception to higher level
+        }
+        List<Column> columns = builder.getTailExpressions();
+        Column head = builder.getHeadExpression();
+        Seq<Column> tails = iterableAsScalaIterable(columns).toSeq();
```

Comment:
If you have a `List<Column> columns` you can do: `columns.get(0)` and `columns.subList(1, columns.size())`

Reply:
Yes, you solution is better